pagopa/io-app
# chore: [IOBP-142] Removal of NativeBase in InitiativeSettings

## Short descriptions
removed NativeBase from InitiativeSettings component
<img width="200" alt="Screenshot 2023-07-21 at 11 32 31" src="https://github.com/pagopa/io-app/assets/60693085/c2d05f97-0bdb-4cd4-bb3a-a1dde048ed78">


## List of changes proposed in this pull request
- moved from custom nativebase component to DS ListItems for the InitiativeSettings component (add card/add Iban ListItems)
- fixed ListItemNav typo that broke the component if passed a ReactElement


## How to test
navigate to a refund-type initiative (non [D] when using dev-server) and make sure both List Items render correctly both in error and not in error, and that when pressed the animation extends to the whole screen's width.
also, check both with and without the experimental DS
